### PR TITLE
P20-308: Create unit tests for course content i18n sync-in

### PR DESCRIPTION
**Pull request by artem-vavilov**
1. Moved `I18n::SyncIn. localize_level_and_project_content ` to `I18n::Resources::Dashboard::CourseContent.sync_in`
2. Created unit tests for `I18n::Resources::Dashboard::CourseContent` module
3. Created unit tests for `I18n::Resources::Dashboard::CourseContent::SyncIn` module

## Links
- jira ticket: [P20-272](https://codedotorg.atlassian.net/browse/P20-272)

## Sync-in
Screenshot 2023-08-10 at 15 29 07

Screenshot 2023-08-10 at 15 29 40

### CODE REVIEW

File: bin/test/i18n/resources/dashboard/course_content/test_sync_in.rb [hunk 1]
```diff
@@ -0,0 +1,711 @@
...
+  #       name: 'Child DSLDefined level',
+  #       display_name: 'Child DSLDefined level display_name',
+  #       bubble_choice_description: 'Child DSLDefined level bubble_choice_description',
+  #       short_instructions: 'Child DSLDefined level short_instructions',
+  #       long_instructions: 'Child DSLDefined level long_instructions',
+  #       # failure_message_override: 'Child DSLDefined level failure_message_overrides',
+  #       teacher_markdown: 'Child DSLDefined level teacher_markdown',
+  #       placeholder: 'Child DSLDefined level placeholder',
+  #       title: 'Child DSLDefined level title',
+  #     )
```
Comment (mgc1194) [inline bin/test/i18n/resources/dashboard/course_content/test_sync_in.rb]: Double checking that this is needed here

Comment (artem-vavilov) [reply bin/test/i18n/resources/dashboard/course_content/test_sync_in.rb]: No, sorry. Will delete this

File: bin/i18n/resources/dashboard/course_content/sync_in.rb [hunk 2]
```diff
@@ -0,0 +1,415 @@
...
+
+            i18n_strings['contained levels'] = level.contained_levels.map do |contained_level|
+              get_i18n_strings(contained_level)
+            end
+
+            i18n_strings.delete_if {|_, value| value.blank?}
+          end
+
+          def write_to_yml(type, strings)
+            File.open(CDO.dir(File.join(I18N_SOURCE_DIR, "dashboard/#{type}.yml")), 'w') do |file|
```
Comment (mgc1194) [inline bin/i18n/resources/dashboard/course_content/sync_in.rb]: Could we add a `FileUtils.mkdir_p` line here? 
The sync fails if running from scratch, i.e if the `i18n/locales/source/dashboard` directory did not exist before.

**Review by mgc1194 — APPROVED** [review]
I was able to run the sync from scratch after adding `FileUtils.mkdir_p(File.join(I18N_SOURCE_DIR, "dashboard"))`, however this would not be necessary if the dashboard directory exists.
Other than that, sync-in was successful:
Screenshot 2023-08-10 at 15 29 07
Screenshot 2023-08-10 at 15 29 40

**Pull request merged**